compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -651,7 +673,7 @@ convertBind env (name, x)
     --
     -- TODO(MH): The check is an approximation which will fail when users
     -- start the name of their own methods with, say, `_exercise`.
-    | any (`T.isPrefixOf` getOccText name) [ "$" <> prefix <> "_" <> method | prefix <- ["dm", "c"], method <- ["create", "fetch", "exercise", "toAnyTemplate", "fromAnyTemplate", "_templateTypeRep", "fetchByKey", "lookupByKey", "toAnyChoice", "fromAnyChoice"] ]
+    | any (`T.isPrefixOf` getOccText name) [ "$" <> prefix <> "_" <> method | prefix <- ["dm", "c"], method <- ["create", "fetch", "exercise", "toAnyTemplate", "fromAnyTemplate", "_templateTypeRep", "fetchByKey", "lookupByKey", "toAnyChoice", "fromAnyChoice", "_toAnyContractKey", "_fromAnyContractKey", "toAnyContractKey", "fromAnyContractKey"] ]
```

Comment:
Why both the `_` prefixed and non-prefixed version? And if that's necessary, then maybe also for `_fromAnyChoice`, `_toAnyChoice`, and `templateTypeRep`?

Reply:
Both seems to be necessary. I’m a bit surprised why it doesn’t seem to be necessary for the AnyChoice methods but I haven’t investigated why it’s not an issue there.

Reply:
I'm surprised we need the version without an `_` prefix here. To me, this feels like we're removing too much. Let's please find out what's going on here before we merge.

Reply:
Fixed, the one we don’t need is the one with the `_`, that is only added to the `TemplateKey` instance but not to the method in the `XInstance` typeclass. I messed this up in `ProposalDesugared` during development which made me think we need to filter both.